TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -307,24 +343,33 @@ private static String formatWorkItem(DiagnosticTrackableCompletableFuture<String
         if (t != null) {
             log.error("Got exception in CompletableFuture callback: ", t);
             requestResponseTriple = new SourceTargetCaptureTuple(rrPair,
-                    new ArrayList<>(), Duration.ZERO
-            );
+                    new ArrayList<>(), new ArrayList<>(),
+                    AggregatedTransformedResponse.HttpRequestTransformationStatus.ERROR, t, Duration.ZERO);
         } else {
             requestResponseTriple = new SourceTargetCaptureTuple(rrPair,
-                    summary.getReceiptTimeAndResponsePackets().map(entry -> entry.getValue()).collect(Collectors.toList()),
+                    summary.requestPackets,
+                    summary.getReceiptTimeAndResponsePackets()
+                            .map(entry -> entry.getValue()).collect(Collectors.toList()),
+                    summary.getTransformationStatus(),
+                    summary.getErrorCause(),
                     summary.getResponseDuration()
             );
         }
 
         try {
+            log.info("Source/Shadow Request/Response tuple: " + requestResponseTriple);
             tripleWriter.writeJSON(requestResponseTriple);
         } catch (IOException e) {
             log.error("Caught an IOException while writing triples output.");
             e.printStackTrace();
             throw new CompletionException(e);
         }
 
-        if (t !=null) { throw new CompletionException(t); }
+        if (t != null) { throw new CompletionException(t); }
```

Comment:
Just now thinking about this, but why don't these exceptions bubble up and cause our app to stop? Is it because of the type of exception or the fact that this method is called within a Consumer function or maybe something else?

Reply:
They're handled with this block in the caller
```
                                } catch (Exception e) {
                                    log.info("base64 gzipped traffic stream: " +
                                            Utils.packetsToCompressedTrafficStream(rrPair.requestData.stream()));
                                    exceptionCount.incrementAndGet();
                                    throw e;
```